### PR TITLE
Fix project created off prem

### PR DESCRIPTION
**Pull request by Andilun**

### CODE REVIEW

File: src/ssb_project_cli/ssb_project/build/poetry.py [hunk 1]
```diff
@@ -101,6 +102,37 @@ def poetry_source_add(
...
+        execute_command(
+            "poetry lock --no-update".split(" "),
+            "source-remove",
+            "Poetry successfully refreshed lock file!",
+            "Poetry failed to refresh lock file.",
+            cwd=cwd,
+        )
+
+
+def nexus_source_not_set_in_lock(source_url: str, cwd: Path) -> bool:
```
Comment (bjornandre) [inline src/ssb_project_cli/ssb_project/build/poetry.py:117]: This was hard to read. I think it is better to call the method something like `exists_in_lockfile` and then use the `not` operator in the if-check. Something like:  `if not exists_in_lockfile(source_url, cwd)`

File: src/ssb_project_cli/ssb_project/build/poetry.py [hunk 2]
```diff
@@ -101,6 +102,37 @@ def poetry_source_add(
...
+    Returns:
+        True if lock nexus source is not set in lock file, else False.
+    """
+    lock_file_path = cwd / Path("poetry.lock")
+    if os.path.isfile(lock_file_path):
+        with open(lock_file_path) as lock_file:
+            if source_url in lock_file.read():
+                return False
+    else:
+        return False
```
[the remark this answers is not in the document]
Comment (Andilun) [reply src/ssb_project_cli/ssb_project/build/poetry.py:132]: When the file does not exist the source is not set.

File: src/ssb_project_cli/ssb_project/build/poetry.py (hunk 1 above)
Comment (Andilun) [reply src/ssb_project_cli/ssb_project/build/poetry.py:117]: The problem with that name(exists_in_lockfile) is that i also want it to return false when the file does not exist.

If the file the lockfile is not created, the source is not set there.

Comment (Andilun) [reply src/ssb_project_cli/ssb_project/build/poetry.py:117]: Is "is_source_not_set_in_lock_file" better?

Comment (bjornandre) [reply src/ssb_project_cli/ssb_project/build/poetry.py]: I think `should_update_lock_file` is less confusing.

Comment (Andilun) [reply src/ssb_project_cli/ssb_project/build/poetry.py]: should_update_lock_file works 👍

**Review by bjornandre — APPROVED** [review]

**Pull request merged**